motoko/invoice-canister/dfx.json

```diff
@@ -1,26 +1,48 @@
 {
-    "canisters": {
-        "invoice": {
-            "dependencies": [
-                "ledger"
-            ],
-            "main": "src/invoice/main.mo",
-            "type": "motoko",
-            "declarations": {
-                "output": "test/e2e/src/declarations"
-            }
-        },
-        "ledger": {
-            "type": "custom",
-            "candid": "src/ledger/ledger.did",
-            "wasm": "src/ledger/ledger.wasm",
-            "remote": {
-                "candid": "src/ledger/ledger.did",
-                "id": {
-                    "ic": "ryjl3-tyaaa-aaaaa-aaaba-cai"
-                }
-            }
+  "canisters": {
+    "invoice": {
+      "declarations": {
+        "output": "test/e2e/src/declarations/invoice",
+        "node_compatibility": true
+      },
+      "main": "src/invoice/Invoice.mo",
+      "type": "motoko"
+    },
+    "nns-ledger": {
+      "build": "",
+      "candid": "candid/nns-ledger.did",
+      "remote": {
+        "id": {
+          "ic": "ryjl3-tyaaa-aaaaa-aaaba-cai",
+          "local": "ryjl3-tyaaa-aaaaa-aaaba-cai"
         }
+      },
+      "type": "custom",
+      "wasm": ""
+    },
+    "icp_ledger_canister": {
+      "type": "custom",
+      "wasm": "src/token-ledger-canisters/icp/ledger.wasm",
+      "candid": "src/token-ledger-canisters/icp/ledger.public.did",
+      "build": ""
+    },
+    "icrc1_token_ledger_canister_ex1": {
```

Comment:
Since we need a couple examples by default, I wonder if it would make sense to use `ckBTC` and `SNS-1`

Reply:
Re: earlier comment: Thanks, it was a great opportunity to get to know core aspects of and developing for the IC. 
 
As far as ckBTC and SNS-1... at the time, the `dfx sns install` wasn't ready to be used along side `dfx nns install`, although it is what I originally wanted to do for an example of an ICRC1 token (using both the `dfx nns` and `dfx sns` commands in the main project). Additionally, because ckBTC/ckTESTBTC requires mainnet connection and local replicas can't connect to the mainnet (for now) , it didn't seem appropriate to add to them to the main project (although I do use the correct values for the ckTESTBTC token canister in the readme as an example of adding a supported token).

For both ckBTC and SNS-1 dedicated examples in the examples/ subdirectory could be more suitable to demonstrate aspects specific to either (particularly if at least mock ckBTC canisters could be deployed to make local replica development possible). This thought is finished in the next reply...

Reply:
Your reasoning makes sense!

I think that any ICRC-1 canister could be deployed as a "mock" `ckBTC` canister for the purposes of local development, apart from the minting and liquidating specifics. Someone may want a more fully fledged mock for those services in their application, but I definitely agree that it's beyond the scope of this repo

Simply by using an ICRC-1 canister like you are currently doing locally, and using the `"remote" -> "id"` config for those canisters in `dfx.json` would probably be sufficient, if we wanted to go that direction